plugins/GSdx/GSState.cpp

```diff
@@ -77,7 +75,18 @@ GSState::GSState()
 	//s_savel = 0;
 
 	UserHacks_WildHack = theApp.GetConfigB("UserHacks") ? theApp.GetConfigI("UserHacks_WildHack") : 0;
-	m_crc_hack_level = theApp.GetConfigI("crc_hack_level");
+
+	GSRendererType renderer = static_cast<GSRendererType>(theApp.GetConfigI("Renderer"));
+	int selected_crc_hack_level = theApp.GetConfigI("crc_hack_level");
+	if (selected_crc_hack_level == -1) {
+		if (renderer == GSRendererType::OGL_HW) {
+			m_crc_hack_level = 2;
+		} else if (renderer == GSRendererType::DX9_HW || renderer == GSRendererType::DX1011_HW) {
+			m_crc_hack_level = 3;
+		}
+	} else {
+		m_crc_hack_level = selected_crc_hack_level;
```

Comment:
Maybe 
```
if (selected_crc_hack_level == -1)
{
    if (renderer == GSRendererType::OGL_HW) {
			m_crc_hack_level = 2;
	} else if (renderer == GSRendererType::DX9_HW || renderer == GSRendererType::DX1011_HW) {
		m_crc_hack_level = 3;
  	}
} else {
		m_crc_hack_level = selected_crc_hack_level;
}
```

**could probably be changed to**

```
if (selected_crc_hack_level == -1)
{
    if (renderer == GSRendererType::OGL_HW)
			m_crc_hack_level = 2;
	else if (renderer == GSRendererType::DX9_HW || renderer == GSRendererType::DX1011_HW)
		    m_crc_hack_level = 3;
 }
else
{
		m_crc_hack_level = selected_crc_hack_level;
}
```

I might be biased but the latter looks much nicer and easier to read, any opinion? If you want to change it to the latter then make sure to also do it for the other files where you had used the previous format.